### ISSUE TITLE
Call super()

### ISSUE THREAD

**Issue by Cyxapic** [microsoft/pyright#238]
**Describe the bug**
When call super in methods it's error if class has mixin

**To Reproduce**
```
class Foo(EggMixin, Bar)
      def method(self, **kwargs):
             data = super().method(**kwargs)  # < - method from Bar, Error: can't accesses to method from EggMixin
             ...
             return data
```

**Expected behavior**
No error

**Comment by erictraut** [comment]
The Python language doesn't have any formal notion of a "mixin class". It simply supports inheritance with a well-defined search order (MRO) when looking up attributes and methods in its parent classes. Pyright follows this same search order when performing static analysis.

I'll need more details from you about how you've defined `EggMixin` and `Bar` in your example above. Does the `EggMixin` class have an instance method called `method`?

Here's a more complete example:
```
class EggMixin:
    def method(self, **kwargs):
        pass

class Bar:
    def method(self, a: int, b: int):
        pass

class Foo(EggMixin, Bar):
      def method(self, **kwargs):
             data = super().method(**kwargs) # This does not generate an error
             data2 = super().method(1, 2) # This correctly generates an error
             return data
```

I've shown two examples of calling super. The first one correctly does not generate an error. The second one correctly does generate an error. So it looks to me like pyright is doing the right thing here. Please share additional details about your case if you still think it's doing something wrong.

**Comment by Cyxapic** [comment]
In my mixin not defined method.
Bar from framework (in my case its django) :)
So i need an "empty method" in mixin? :)

**Comment by erictraut** [comment]
So your example looks like the following?

```
class EggMixin:
    # No "method" defined here
    pass

class Bar:
    def method(self, **kwargs):
        pass

class Foo(EggMixin, Bar):
      def method(self, **kwargs):
             data = super().method(**kwargs)
             return data
```

This works fine for me in pyright. No error generated. There must still be something more to your problem. Please post a minimal (but complete) repro.

**Comment by erictraut** [comment]
Closing this since I haven't heard a response. If you still think this is a problem, please respond, and I'll re-open.

**Comment by psam44** [comment]
I confirm it is not a problem with Pyright, but a lack of typing information.

The OP didn't stress enough that the so-called Bar class doesn't come from his code but from a framework (Django in this case). For a full type resolution, the stub files of the framework have to be installed. These stubs are available at https://pypi.org/project/django-stubs/.

**Comment by psam44** [comment]
@erictraut Could you please investigate this case, elaborated from your code sample:

```python
class EggMixin:
    # No "method" defined here
    pass

# from tmp import Bar  # still working, as a sibling
from tmp_for_pyright.tmp import Bar  # no more working, if in another package

class Foo(EggMixin, Bar):
      def method(self, **kwargs):
             data = super().method(**kwargs)
             return data
```
with Bar deported in another package, such as D:\Python36\Lib\site-packages\tmp_for_pyright\tmp.py

```python
class Bar:
    def method(self, **kwargs):
        pass
```

The error:
```
Cannot access member "method" for type "EggMixin"
  Member "method" is unknown  Pyright (reportGeneralTypeIssues)
```

**Comment by erictraut** [comment]
It works fine for me in this case. I suspect that the import is not being resolved correctly. If you hover over "Bar", does it indicate that it's a class, or an "Unknown" type?

Normally an import resolution failure would result in an error. Do you have the "reportMissingImports" diagnostic disabled? 

Here's a screen shot of what it looks like when the import is resolved correctly and a second screen shot when the import resolution fails.

![Screen Shot 2020-08-29 at 1 42 22 PM](https://user-images.githubusercontent.com/7040122/91645818-af950c80-e9fd-11ea-960f-cf08e5f75a30.png)
![Screen Shot 2020-08-29 at 1 42 41 PM](https://user-images.githubusercontent.com/7040122/91645819-b15ed000-e9fd-11ea-8edb-6c23d359be1b.png)

**Comment by psam44** [comment]
Your test is my: # from tmp import Bar  # still working, as a sibling
You should try the one with an import from another package.
I don't have any import problem.

**Comment by erictraut** [comment]
That works fine for me too. I suspect that the type of "Bar" is unknown. If you hover over "Bar", does it indicate that it's a class, or an "Unknown" type?

**Comment by psam44** [comment]
![image](https://user-images.githubusercontent.com/537094/91646016-2d333980-ea4b-11ea-8881-41f22a790c57.png)

**Comment by erictraut** [comment]
That's the problem. You need to determine why Bar is an unknown type and fix that.

**Comment by psam44** [comment]
If it has a relation, my setting "Use Library Code for Types" is unchecked.
Probably ... I just checked it and:
![image](https://user-images.githubusercontent.com/537094/91646183-d62e6400-ea4c-11ea-9b5c-7d74dc41859b.png)

but it brings too many other errors on my base code, that's why I prefer it unchecked. Anyway, to my understanding, this setting is not mandatory, unknown things are just unknown, not an error.

**Comment by psam44** [comment]
For comparison, it works anyway, with the setting unchecked, in this case (Bar is first parent):
![image](https://user-images.githubusercontent.com/537094/91646318-22c66f00-ea4e-11ea-90bc-e069c8908980.png)

**Comment by erictraut** [comment]
Yes, you are right. It really should not emit an error in this case. I've addressed the issue, and the fix will be in the next release.
